### PR TITLE
Do not generate internal archives

### PR DESCRIPTION
**Pull request by mmitche**

### CODE REVIEW

**Comment by mmitche** [comment]
Start of https://github.com/dotnet/runtime/issues/44476

**Review by hoyosjs — APPROVED** [review]

**Comment by mmitche** [comment]
https://dev.azure.com/dnceng/internal/_build/results?buildId=2237877&view=logs&j=89bc6db5-401c-5796-8771-4042af6a0738&t=8f6ff74d-d6fe-5959-a3fe-ddb76bfdded4&l=4267

**Comment by hoyosjs** [comment]
As long as the needed archives are produced.

File: src/installer/pkg/sfx/Microsoft.NETCore.App/Microsoft.NETCore.App.MonoCrossAOT.sfxproj [hunk 1]
```diff
@@ -22,6 +22,10 @@
     <NativeRuntimeAsset Condition="!$([MSBuild]::IsOsPlatform('Windows'))" Include="$(IntermediateOutputPath)Microsoft.NETCore.App.MonoCrossAOT.UnixFilePermissions.xml" TargetPath="data/UnixFilePermissions.xml" />
   </ItemGroup>
 
+  <ItemGroup>
+    <PackageReference Include="Microsoft.DotNet.Build.Tasks.Archives" Version="$(MicrosoftDotNetBuildTasksArchivesVersion)" />
```
[the remark this answers is not in the document]
Comment (mmitche) [reply src/installer/pkg/sfx/Microsoft.NETCore.App/Microsoft.NETCore.App.MonoCrossAOT.sfxproj:26]: It shouldn't have gotten rid of that archive. Surprised at that one

[the remark this answers is not in the document]
Comment (hoyosjs) [reply src/installer/pkg/sfx/Microsoft.NETCore.App/Microsoft.NETCore.App.MonoCrossAOT.sfxproj:26]: Did this drop ever exist: /__w/1/s/artifacts/packages/Release/Shipping//dotnet-monocrossaot-8.0.0-ci-linux-x64.tar.gz

**Comment by mmitche** [comment]
https://dev.azure.com/dnceng/internal/_build/results?buildId=2237975&view=results

**Comment by mmitche** [comment]
https://dev.azure.com/dnceng/internal/_build/results?buildId=2237991&view=results

**Comment by mmitche** [comment]
Official build looking good. Appears to remove about 3GB in artifacts.

**Comment by mmitche** [comment]
> Official build looking good. Appears to remove about 3GB in artifacts.

Doesn't look like it's quite that big. Pushed another commit to avoid removal of the symbols archives, and the current main build has signing off so it's not a fair comparison.

**Comment by mmitche** [comment]
@hoyosjs Any idea whether the symbols archives are used today? There isn't a way to create the symbols archive without creating the internal archive without arcade changes.

**Comment by hoyosjs** [comment]
Not uses that I know of.

**Comment by mmitche** [comment]
Looks like source-build may be a consumer...

https://github.com/dotnet/runtime/issues/49026

**Comment by jkoritzinsky** [comment]
I’m not sure if source-build is a consumer or if source-build was just expecting the artifacts list to be the same.

Maybe our source-build partners use the artifacts to build symbol debs/rpms?

**Comment by mmitche** [comment]
> I’m not sure if source-build is a consumer or if source-build was just expecting the artifacts list to be the same.
> 
> Maybe our source-build partners use the artifacts to build symbol debs/rpms?

@omajid What do the symbol archives get used for in your ecosystem? Do they get distributed to customers? They're not as useful to MSFT since the symbols get published to the symbol servers, and are archived in .symbols.nupkg files.

/cc @MichaelSimons

**Comment by mmitche** [comment]
We don't produce the symbol packages, as noted in https://github.com/dotnet/source-build/issues/3547. I propose that we eliminate the symbol archives **except** in source-build to eliminate duplicate data. Then, we work to get the .symbols.nupkg files working source-build, which will enable symbols for the shared frameworks and nuget packages, like @NikolaMilosavljevic mentions in the issue.

At that point we could eliminate the old symbol archives.

Thoughts?

**Comment by mmitche** [comment]
Alright, I think this is good to go.  Removed artifacts are below. In source build configs, the symbols archives are still produced.

```
dotnet-hostfxr-internal-8.0.0-rc.1.23408.5-linux-arm.tar.gz
dotnet-hostfxr-internal-8.0.0-rc.1.23408.5-linux-arm.tar.gz.sha512
dotnet-hostfxr-internal-8.0.0-rc.1.23408.5-linux-arm64.tar.gz
dotnet-hostfxr-internal-8.0.0-rc.1.23408.5-linux-arm64.tar.gz.sha512
dotnet-hostfxr-internal-8.0.0-rc.1.23408.5-linux-bionic-arm64.tar.gz
dotnet-hostfxr-internal-8.0.0-rc.1.23408.5-linux-bionic-arm64.tar.gz.sha512
dotnet-hostfxr-internal-8.0.0-rc.1.23408.5-linux-bionic-x64.tar.gz
dotnet-hostfxr-internal-8.0.0-rc.1.23408.5-linux-bionic-x64.tar.gz.sha512
dotnet-hostfxr-internal-8.0.0-rc.1.23408.5-linux-musl-arm.tar.gz
dotnet-hostfxr-internal-8.0.0-rc.1.23408.5-linux-musl-arm.tar.gz.sha512
dotnet-hostfxr-internal-8.0.0-rc.1.23408.5-linux-musl-arm64.tar.gz
dotnet-hostfxr-internal-8.0.0-rc.1.23408.5-linux-musl-arm64.tar.gz.sha512
dotnet-hostfxr-internal-8.0.0-rc.1.23408.5-linux-musl-x64.tar.gz
dotnet-hostfxr-internal-8.0.0-rc.1.23408.5-linux-musl-x64.tar.gz.sha512
dotnet-hostfxr-internal-8.0.0-rc.1.23408.5-linux-x64.tar.gz
dotnet-hostfxr-internal-8.0.0-rc.1.23408.5-linux-x64.tar.gz.sha512
dotnet-hostfxr-internal-8.0.0-rc.1.23408.5-osx-arm64.tar.gz
dotnet-hostfxr-internal-8.0.0-rc.1.23408.5-osx-arm64.tar.gz.sha512
dotnet-hostfxr-internal-8.0.0-rc.1.23408.5-osx-x64.tar.gz
dotnet-hostfxr-internal-8.0.0-rc.1.23408.5-osx-x64.tar.gz.sha512
dotnet-hostfxr-internal-8.0.0-rc.1.23408.5-win-arm64.zip
dotnet-hostfxr-internal-8.0.0-rc.1.23408.5-win-arm64.zip.sha512
dotnet-hostfxr-internal-8.0.0-rc.1.23408.5-win-x64.zip
dotnet-hostfxr-internal-8.0.0-rc.1.23408.5-win-x64.zip.sha512
dotnet-hostfxr-internal-8.0.0-rc.1.23408.5-win-x86.zip
dotnet-hostfxr-internal-8.0.0-rc.1.23408.5-win-x86.zip.sha512
dotnet-hostfxr-internal-pgo-8.0.0-rc.1.23408.5-linux-arm64.tar.gz
dotnet-hostfxr-internal-pgo-8.0.0-rc.1.23408.5-linux-arm64.tar.gz.sha512
dotnet-hostfxr-internal-pgo-8.0.0-rc.1.23408.5-linux-x64.tar.gz
dotnet-hostfxr-internal-pgo-8.0.0-rc.1.23408.5-linux-x64.tar.gz.sha512
dotnet-hostfxr-internal-pgo-8.0.0-rc.1.23408.5-win-arm64.zip
dotnet-hostfxr-internal-pgo-8.0.0-rc.1.23408.5-win-arm64.zip.sha512
dotnet-hostfxr-internal-pgo-8.0.0-rc.1.23408.5-win-x64.zip
dotnet-hostfxr-internal-pgo-8.0.0-rc.1.23408.5-win-x64.zip.sha512
dotnet-hostfxr-internal-pgo-8.0.0-rc.1.23408.5-win-x86.zip
dotnet-hostfxr-internal-pgo-8.0.0-rc.1.23408.5-win-x86.zip.sha512
dotnet-runtime-internal-8.0.0-rc.1.23408.5-linux-arm.tar.gz
dotnet-runtime-internal-8.0.0-rc.1.23408.5-linux-arm.tar.gz.sha512
dotnet-runtime-internal-8.0.0-rc.1.23408.5-linux-arm64.tar.gz
dotnet-runtime-internal-8.0.0-rc.1.23408.5-linux-arm64.tar.gz.sha512
dotnet-runtime-internal-8.0.0-rc.1.23408.5-linux-bionic-arm64.tar.gz
dotnet-runtime-internal-8.0.0-rc.1.23408.5-linux-bionic-arm64.tar.gz.sha512
dotnet-runtime-internal-8.0.0-rc.1.23408.5-linux-bionic-x64.tar.gz
dotnet-runtime-internal-8.0.0-rc.1.23408.5-linux-bionic-x64.tar.gz.sha512
dotnet-runtime-internal-8.0.0-rc.1.23408.5-linux-musl-arm.tar.gz
dotnet-runtime-internal-8.0.0-rc.1.23408.5-linux-musl-arm.tar.gz.sha512
dotnet-runtime-internal-8.0.0-rc.1.23408.5-linux-musl-arm64.tar.gz
dotnet-runtime-internal-8.0.0-rc.1.23408.5-linux-musl-arm64.tar.gz.sha512
dotnet-runtime-internal-8.0.0-rc.1.23408.5-linux-musl-x64.tar.gz
dotnet-runtime-internal-8.0.0-rc.1.23408.5-linux-musl-x64.tar.gz.sha512
dotnet-runtime-internal-8.0.0-rc.1.23408.5-linux-x64.tar.gz
dotnet-runtime-internal-8.0.0-rc.1.23408.5-linux-x64.tar.gz.sha512
dotnet-runtime-internal-8.0.0-rc.1.23408.5-osx-arm64.tar.gz
dotnet-runtime-internal-8.0.0-rc.1.23408.5-osx-arm64.tar.gz.sha512
dotnet-runtime-internal-8.0.0-rc.1.23408.5-osx-x64.tar.gz
dotnet-runtime-internal-8.0.0-rc.1.23408.5-osx-x64.tar.gz.sha512
dotnet-runtime-internal-8.0.0-rc.1.23408.5-win-arm64.zip
dotnet-runtime-internal-8.0.0-rc.1.23408.5-win-arm64.zip.sha512
dotnet-runtime-internal-8.0.0-rc.1.23408.5-win-x64.zip
dotnet-runtime-internal-8.0.0-rc.1.23408.5-win-x64.zip.sha512
dotnet-runtime-internal-8.0.0-rc.1.23408.5-win-x86.zip
dotnet-runtime-internal-8.0.0-rc.1.23408.5-win-x86.zip.sha512
dotnet-runtime-internal-pgo-8.0.0-rc.1.23408.5-linux-arm64.tar.gz
dotnet-runtime-internal-pgo-8.0.0-rc.1.23408.5-linux-arm64.tar.gz.sha512
dotnet-runtime-internal-pgo-8.0.0-rc.1.23408.5-linux-x64.tar.gz
dotnet-runtime-internal-pgo-8.0.0-rc.1.23408.5-linux-x64.tar.gz.sha512
dotnet-runtime-internal-pgo-8.0.0-rc.1.23408.5-win-arm64.zip
dotnet-runtime-internal-pgo-8.0.0-rc.1.23408.5-win-arm64.zip.sha512
dotnet-runtime-internal-pgo-8.0.0-rc.1.23408.5-win-x64.zip
dotnet-runtime-internal-pgo-8.0.0-rc.1.23408.5-win-x64.zip.sha512
dotnet-runtime-internal-pgo-8.0.0-rc.1.23408.5-win-x86.zip
dotnet-runtime-internal-pgo-8.0.0-rc.1.23408.5-win-x86.zip.sha512
dotnet-runtime-symbols-linux-arm-8.0.0-rc.1.23408.5.tar.gz
dotnet-runtime-symbols-linux-arm-8.0.0-rc.1.23408.5.tar.gz.sha512
dotnet-runtime-symbols-linux-arm64-8.0.0-rc.1.23408.5.tar.gz
dotnet-runtime-symbols-linux-arm64-8.0.0-rc.1.23408.5.tar.gz.sha512
dotnet-runtime-symbols-linux-bionic-arm64-8.0.0-rc.1.23408.5.tar.gz
dotnet-runtime-symbols-linux-bionic-arm64-8.0.0-rc.1.23408.5.tar.gz.sha512
dotnet-runtime-symbols-linux-bionic-x64-8.0.0-rc.1.23408.5.tar.gz
dotnet-runtime-symbols-linux-bionic-x64-8.0.0-rc.1.23408.5.tar.gz.sha512
dotnet-runtime-symbols-linux-musl-arm-8.0.0-rc.1.23408.5.tar.gz
dotnet-runtime-symbols-linux-musl-arm-8.0.0-rc.1.23408.5.tar.gz.sha512
dotnet-runtime-symbols-linux-musl-arm64-8.0.0-rc.1.23408.5.tar.gz
dotnet-runtime-symbols-linux-musl-arm64-8.0.0-rc.1.23408.5.tar.gz.sha512
dotnet-runtime-symbols-linux-musl-x64-8.0.0-rc.1.23408.5.tar.gz
dotnet-runtime-symbols-linux-musl-x64-8.0.0-rc.1.23408.5.tar.gz.sha512
dotnet-runtime-symbols-linux-x64-8.0.0-rc.1.23408.5.tar.gz
dotnet-runtime-symbols-linux-x64-8.0.0-rc.1.23408.5.tar.gz.sha512
dotnet-runtime-symbols-osx-arm64-8.0.0-rc.1.23408.5.tar.gz
dotnet-runtime-symbols-osx-arm64-8.0.0-rc.1.23408.5.tar.gz.sha512
dotnet-runtime-symbols-osx-x64-8.0.0-rc.1.23408.5.tar.gz
dotnet-runtime-symbols-osx-x64-8.0.0-rc.1.23408.5.tar.gz.sha512
dotnet-runtime-symbols-win-arm64-8.0.0-rc.1.23408.5.zip
dotnet-runtime-symbols-win-arm64-8.0.0-rc.1.23408.5.zip.sha512
dotnet-runtime-symbols-win-x64-8.0.0-rc.1.23408.5.zip
dotnet-runtime-symbols-win-x64-8.0.0-rc.1.23408.5.zip.sha512
dotnet-runtime-symbols-win-x86-8.0.0-rc.1.23408.5.zip
dotnet-runtime-symbols-win-x86-8.0.0-rc.1.23408.5.zip.sha512
dotnet-targeting-pack-8.0.0-rc.1.23408.5-linux-arm.tar.gz
dotnet-targeting-pack-8.0.0-rc.1.23408.5-linux-arm.tar.gz.sha512
dotnet-targeting-pack-8.0.0-rc.1.23408.5-linux-arm64.tar.gz
dotnet-targeting-pack-8.0.0-rc.1.23408.5-linux-arm64.tar.gz.sha512
dotnet-targeting-pack-8.0.0-rc.1.23408.5-linux-bionic-arm64.tar.gz
dotnet-targeting-pack-8.0.0-rc.1.23408.5-linux-bionic-arm64.tar.gz.sha512
dotnet-targeting-pack-8.0.0-rc.1.23408.5-linux-bionic-x64.tar.gz
dotnet-targeting-pack-8.0.0-rc.1.23408.5-linux-bionic-x64.tar.gz.sha512
dotnet-targeting-pack-8.0.0-rc.1.23408.5-linux-musl-arm.tar.gz
dotnet-targeting-pack-8.0.0-rc.1.23408.5-linux-musl-arm.tar.gz.sha512
dotnet-targeting-pack-8.0.0-rc.1.23408.5-linux-musl-arm64.tar.gz
dotnet-targeting-pack-8.0.0-rc.1.23408.5-linux-musl-arm64.tar.gz.sha512
dotnet-targeting-pack-8.0.0-rc.1.23408.5-linux-musl-x64.tar.gz
dotnet-targeting-pack-8.0.0-rc.1.23408.5-linux-musl-x64.tar.gz.sha512
dotnet-targeting-pack-8.0.0-rc.1.23408.5-linux-x64.tar.gz
dotnet-targeting-pack-8.0.0-rc.1.23408.5-linux-x64.tar.gz.sha512
dotnet-targeting-pack-8.0.0-rc.1.23408.5-osx-arm64.tar.gz
dotnet-targeting-pack-8.0.0-rc.1.23408.5-osx-arm64.tar.gz.sha512
dotnet-targeting-pack-8.0.0-rc.1.23408.5-osx-x64.tar.gz
dotnet-targeting-pack-8.0.0-rc.1.23408.5-osx-x64.tar.gz.sha512
dotnet-targeting-pack-8.0.0-rc.1.23408.5-win-arm64.zip
dotnet-targeting-pack-8.0.0-rc.1.23408.5-win-arm64.zip.sha512
dotnet-targeting-pack-8.0.0-rc.1.23408.5-win-x64.zip
dotnet-targeting-pack-8.0.0-rc.1.23408.5-win-x64.zip.sha512
dotnet-targeting-pack-8.0.0-rc.1.23408.5-win-x86.zip
dotnet-targeting-pack-8.0.0-rc.1.23408.5-win-x86.zip.sha512
```

**Review by jkoritzinsky — APPROVED** [review]

**Pull request merged**